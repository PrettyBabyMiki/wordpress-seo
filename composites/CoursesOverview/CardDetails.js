import React, { Fragment } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { __ } from "@wordpress/i18n";
import colors from "../../style-guide/colors.json";
import { makeOutboundLink } from "../../utils/makeOutboundLink";

const CardLinkButton = styled.a`
	align-items: center;
	justify-content: center;
	vertical-align: middle;
	cursor: pointer;
	color: ${ colors.$color_black };
	white-space: nowrap;
	display: block;
	border-radius: 4px;
	background-color: ${ colors.$color_button_upsell };
	padding: 12px 16px;
	box-shadow: inset 0 -4px 0 rgba(0, 0, 0, 0.2);
	border: none;
	text-decoration: none;
	font-weight: bold;
	font-size: inherit;
	margin-top: 24px;

	&:hover,
	&:focus,
	&:active {
		color: ${ colors.$color_black };
		background: ${ colors.$color_button_upsell_hover };
	}

	:active {
		position: relative;
		top: 2px;
		background-color: ${ colors.$color_button_hover_upsell };
		box-shadow: none;
		filter: none;
	}
`;

const CardInfoLink = styled.a`
	padding: 12px 16px;
	font-weight: bold;
`;

const OutboundLinkButton = makeOutboundLink( CardLinkButton );
const OutboundInfoLink = makeOutboundLink( CardInfoLink );

const ActionBlock = styled.div`
	text-align: center;
`;

const Header = styled.a`
	padding: 0;
	margin: 0;
	margin-bottom: 15px;

	color: ${ colors.$color_pink_dark };
	font-weight: 300;
	font-size: 1.5em;
	text-decoration: none;
`;

const Details = styled.div`
	margin-bottom:24px;
	border-bottom: 1px ${ colors.$color_grey } solid;
	flex-grow: 1;
`;

/**
 * @summary CardDetails component.
 *
 * @returns {Component} The rendered CardDetails component.
 */
class CardDetails extends React.Component {
	/**
	 * Sets the CourseCard object.
	 *
	 * @returns {void}
	 */
	constructor() {
		super();
	}

	/**
	 * Renders the component.
	 *
	 * @returns {ReactElement} The rendered component.
	 */
	render() {
		return (
			<Fragment>
				<Details>
					<Header href={ this.props.courseUrl }>{ this.props.title }</Header>
					<p>{ this.props.description }</p>
				</Details>
				<ActionBlock>
					<OutboundInfoLink href={ this.props.courseUrl }>
						{
							__( "Read more about this course", "wordpress-seo" )
						}
					</OutboundInfoLink>
					<OutboundLinkButton href={ this.props.shopUrl }>
						{
							__( "Get the full course", "wordpress-seo" )
						}
					</OutboundLinkButton>
				</ActionBlock>
			</Fragment>
		);
	}
}

export default CardDetails;

CardDetails.propTypes = {
	title: PropTypes.string,
	description: PropTypes.string,
	courseUrl: PropTypes.string,
	shopUrl: PropTypes.string,
};

CardDetails.defaultProps = {
	title: null,
	description: null,
	courseUrl: null,
	shopUrl: null,
};
