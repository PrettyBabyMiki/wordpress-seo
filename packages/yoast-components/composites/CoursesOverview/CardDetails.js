export { CardDetails as default } from "@yoast/components";
