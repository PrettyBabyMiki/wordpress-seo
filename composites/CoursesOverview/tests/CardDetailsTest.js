import React from "react";
import renderer from "react-test-renderer";
import CardDetails from "../CardDetails";

test( "The empty CardDetails component matches the snapshot", () => {
	const component = renderer.create(
		<CardDetails />
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "The CardDetails component matches the snapshot when given specific props", () => {
	const course = {
		title: "Keyword Research",
		description: "Do you know the essential first step of good SEO? It’s keyword research. In this training, \n" +
		             "you’ll learn how to research and select the keywords that will guide searchers to your pages.",
		courseUrl: "https://yoast.com/academy/keyword-research-training/",
		shopUrl: "https://yoast.com/cart/?add-to-cart=1311259",
	};

	const component = renderer.create(
		<CardDetails
			title={ course.title }
			description={ course.description }
			courseUrl={ course.courseUrl }
			shopUrl={ course.shopUrl }
		/>
	);

	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
