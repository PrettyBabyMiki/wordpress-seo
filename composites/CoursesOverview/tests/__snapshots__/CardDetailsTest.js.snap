// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CardDetails component matches the snapshot when given specific props 1`] = `
Array [
  .c3 li {
  position: relative;
  margin-left: 16px;
}

.c3 li:before {
  content: "✓";
  color: #77b227;
  position: absolute;
  font-weight: bold;
  display: inline-block;
  left: -16px;
}

.c1 {
  margin: 0 0 8px 0;
  display: block;
  font-weight: 400;
  font-size: 1.5em;
  line-height: 1.2;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #a4286a;
}

.c0 {
  margin-bottom: 12px;
  border-bottom: 1px #ddd solid;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
    className="c0"
  >
    <h2
      className="c1"
    >
      <a
        className="c2"
        href="https://yoast.com/academy/keyword-research-training/"
      >
        Keyword Research
      </a>
    </h2>
    <div
      className="c3"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Do you know the essential first step of good SEO? It’s keyword research. In this training, 
you’ll learn how to research and select the keywords that will guide searchers to your pages.",
        }
      }
    />
  </div>,
  .c2 {
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute !important;
  width: 1px;
  word-wrap: normal !important;
  -webkit-transform: translateY(1em);
  -ms-transform: translateY(1em);
  transform: translateY(1em);
}

.c1 {
  cursor: pointer;
  color: #000;
  white-space: nowrap;
  display: block;
  border-radius: 4px;
  background-color: #fec228;
  padding: 12px 16px;
  box-shadow: inset 0 -4px 0 rgba(0,0,0,0.2);
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: bold;
  font-size: inherit;
  margin-top: 0;
  margin-bottom: 8px;
}

.c1:hover,
.c1:focus,
.c1:active {
  color: #000;
  background: #f2ae01;
}

.c1:active {
  position: relative;
  top: 2px;
  box-shadow: none;
  -webkit-filter: none;
  filter: none;
}

.c3 {
  font-weight: bold;
}

.c0 {
  text-align: center;
}

<div
    className="c0"
  >
    <a
      className="c1"
      href="https://yoast.com/cart/?add-to-cart=1311259"
      rel="noopener noreferrer"
      target="_blank"
    >
      Get the full course
      <span
        className="c2"
      >
        (Opens in a new browser tab)
      </span>
    </a>
    <a
      className="c3"
      href="https://yoast.com/academy/keyword-research-training/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Read more about this course
      <span
        className="c2"
      >
        (Opens in a new browser tab)
      </span>
    </a>
  </div>,
]
`;

exports[`The empty CardDetails component matches the snapshot 1`] = `
Array [
  .c3 li {
  position: relative;
  margin-left: 16px;
}

.c3 li:before {
  content: "✓";
  color: #77b227;
  position: absolute;
  font-weight: bold;
  display: inline-block;
  left: -16px;
}

.c1 {
  margin: 0 0 8px 0;
  display: block;
  font-weight: 400;
  font-size: 1.5em;
  line-height: 1.2;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #a4286a;
}

.c0 {
  margin-bottom: 12px;
  border-bottom: 1px #ddd solid;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
    className="c0"
  >
    <h2
      className="c1"
    >
      <a
        className="c2"
        href=""
      >
        
      </a>
    </h2>
    <div
      className="c3"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </div>,
  .c2 {
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute !important;
  width: 1px;
  word-wrap: normal !important;
  -webkit-transform: translateY(1em);
  -ms-transform: translateY(1em);
  transform: translateY(1em);
}

.c1 {
  cursor: pointer;
  color: #000;
  white-space: nowrap;
  display: block;
  border-radius: 4px;
  background-color: #fec228;
  padding: 12px 16px;
  box-shadow: inset 0 -4px 0 rgba(0,0,0,0.2);
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: bold;
  font-size: inherit;
  margin-top: 0;
  margin-bottom: 8px;
}

.c1:hover,
.c1:focus,
.c1:active {
  color: #000;
  background: #f2ae01;
}

.c1:active {
  position: relative;
  top: 2px;
  box-shadow: none;
  -webkit-filter: none;
  filter: none;
}

.c3 {
  font-weight: bold;
}

.c0 {
  text-align: center;
}

<div
    className="c0"
  >
    <a
      className="c1"
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      Get the full course
      <span
        className="c2"
      >
        (Opens in a new browser tab)
      </span>
    </a>
    <a
      className="c3"
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      Read more about this course
      <span
        className="c2"
      >
        (Opens in a new browser tab)
      </span>
    </a>
  </div>,
]
`;
